code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -226,10 +236,75 @@
 	else
 		return
 
+/mob/living/simple_animal/bot/medbot/proc/tip_over(mob/user)
+	playsound(src, 'sound/machines/warning-buzzer.ogg', 50)
+	user.visible_message("<span class='danger'>[user] tips over [src]!</span>", "<span class='danger'>You tip [src] over!</span>")
+	mode = BOT_TIPPED
+	var/matrix/mat = transform
+	transform = mat.Turn(180)
+	tipper_name = user.name
+
+/mob/living/simple_animal/bot/medbot/proc/set_right(mob/user)
+	if(user)
+		playsound(src, 'sound/machines/ping.ogg', 50)
+		user.visible_message("<span class='notice'>[user] sets [src] right-side up!</span>", "<span class='green'>You set [src] right-side up!</span>")
+		speak("Thank you!")
+	else
+		visible_message("<span class='notice'>[src] manages to writhe around enough to right itself.</span>")
+		tipper_name = null
+	tipped_status = MEDBOT_PANIC_NONE
+	mode = BOT_IDLE
+	transform = matrix()
+
+/// if someone tipped us over, check whether we should ask for help or just right ourselves eventually
+/mob/living/simple_animal/bot/medbot/proc/handle_panic()
+	tipped_status++
+	switch(tipped_status)
+		if(MEDBOT_PANIC_LOW)
+			speak("Please put me back.")
+		if(MEDBOT_PANIC_MED)
+			speak("Hey, I need help! Please help me up!")
+		if(MEDBOT_PANIC_HIGH)
+			speak("Please help! This is scaring me!!")
+		if(MEDBOT_PANIC_FUCK)
+			speak("I DON'T LIKE BEING TIPPED! PLEASE HELP ME!!")
+		if(MEDBOT_PANIC_ENDING)
+			speak("THIS HURTS, PLEASE HELP ME!!")
+		if(MEDBOT_PANIC_END)
+			speak("PSYCH ALERT: Crewmember [tipper_name] recorded displaying antisocial tendencies torturing bots in [get_area(src)]. Please schedule psych evaluation.", radio_channel)

```

Comment:
it's a bit weird that the medibot immediately goes from screaming in panic to totally fine (and even calmly making a psych report about its traumatic experience) immediately after it rights itself.

also, you might want to add a word between "tendencies" and "torturing".